### PR TITLE
CLDC-771: Allow Data Coordinator to change details for users in their organisation

### PR DESCRIPTION
**Pull request by baarkerlounger**
For users in their organisation Data Coordinators can:

- Update name
- Update email
- Update role

They cannot update password (only the user themselves can do that). (Though they technically could update email and use that to reset password).

Data coordinator viewing users account:
image

Data coordinator editing users account:
image

Data coordinator viewing their own account:
image

Data coordinator editing their own account:
image

### CODE REVIEW

**Review by kosiakkatrina — APPROVED** [review]

File: app/views/users/edit.html.erb [hunk 1]
```diff
@@ -26,6 +26,10 @@
         spellcheck: "false"
       %>
 
+      <%= f.govuk_check_boxes_fieldset :is_dpo, multiple: false, legend: { text: "Are #{@user == current_user ? "you" : "they"} a data protection officer?", size: "m" } do %>
+        <%= f.govuk_check_box :is_dpo, 1, 0, multiple: false, link_errors: true, label: { text: "Yes" } %>
+      <% end %>
+
```
Comment (paulrobertlloyd) [inline app/views/users/edit.html.erb]: Use Yes/No radios rather than a checkbox

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -31,6 +31,10 @@
         f.govuk_collection_radio_buttons :role, roles, :id, :name, legend: { text: "Role", size: "m" }
       %>
 
+      <%= f.govuk_check_boxes_fieldset :is_dpo, legend: { text: "Are they a data protection officer?", size: "m" } do %>
+        <%= f.govuk_check_box :is_dpo, :is_dpo, label: { text: "Yes" } %>
+      <% end %>
+
```
Comment (paulrobertlloyd) [inline app/views/users/new.html.erb]: Ditto

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -13,33 +13,43 @@
...
         row.key { 'Role' }
-        row.value { current_user.role.humanize }
+        row.value { @user.role.humanize }
         row.action()
       end %>
+
+      <%= summary_list.row do |row|
+        row.key { 'Data protection officer' }
+        row.value { @user.is_data_protection_officer? ? "Yes" : "No" }
+        row.action(visually_hidden_text: 'are they a data protection officer?', href: edit_user_path, html_attributes: { 'data-qa': 'change-are-they-a-data-protection-officer' })
```
Comment (paulrobertlloyd) [inline app/views/users/show.html.erb]: Given that the visible label is ‘Change’, probably makes sense for this to say:

```suggestion
        row.action(visually_hidden_text: 'if they a data protection officer', href: edit_user_path, html_attributes: { 'data-qa': 'change-are-they-a-data-protection-officer' })
```

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]

File: app/views/users/show.html.erb (hunk 3 above)
Comment (baarkerlounger) [reply app/views/users/show.html.erb]: Change "whether" they are a data protection officer? @Dan13L3 what's your preference?

File: app/views/users/edit.html.erb (hunk 1 above)
Comment (baarkerlounger) [reply app/views/users/edit.html.erb]: Even though the answer will hardly ever be yes?

Comment (baarkerlounger) [reply app/views/users/edit.html.erb]: Done

File: app/views/users/new.html.erb (hunk 2 above)
Comment (baarkerlounger) [reply app/views/users/new.html.erb]: Done

File: app/views/users/show.html.erb (hunk 3 above)
Comment (Dan13L3) [reply app/views/users/show.html.erb]: @baarkerlounger Can it be dynamic, so if someone's changing someone else's details, it's 'Are they a data protection officer?', and if it's their own, 'Are you a data protection officer?' (And then the label under personal details stays as 'Data protection officer'.)

**Pull request merged**